src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsUrlTest.java

```diff
@@ -0,0 +1,200 @@
+                        "URL in" + getXmlPath() + " is inconsistent: "
+                                + sub[0] + " " + sub[1] + " " + sub[2]);
+            }
+            else {
+                final String[] moduleFileNames = checksNamesMap.get(sub[0]);
+                if (moduleFileNames != null) {
+                    boolean match = false;
+                    for (String filename : moduleFileNames) {
+                        if ((sub[1] + SUFFIX_CHECK_JAVA).equals(filename)) {
+                            match = true;
+                            break;
+                        }
+                    }
+                    assertTrue(match, "URL in" + getXmlPath() + " is inconsistent: "
+                            + sub[0] + " " + sub[1] + " " + sub[2]);
+                }
+            }
+        }
+    }
+
+    public static final class CheckTest extends DefaultHandler {
+
+        public static final String SPLIT_ONE = "#";
+
+        public static final String SPLIT_TWO = "\\.";
+
+        public static final String PREFIX_CONFIG = "config_";
+
+        private final String nodeName;
+
+        private String[] strings;
+
+        private List<String[]> list;
+
+        private String currentTag;
+
+        public CheckTest(String nodeName) {
+            this.nodeName = nodeName;
+        }
+
+        public String[][] getStrings() {
+            final String[][] stringArrays = new String[list.size()][];
+            return list.toArray(stringArrays);
+        }
+
+        @Override
+        public void startDocument() {
+            list = new ArrayList<>();
+        }
+
+        @Override
+        public void startElement(String uri, String localName, String qName,
+                                 Attributes attributes) {
+            if (qName.equals(nodeName)) {
+                strings = new String[3];
+                final String[] splitStrings = attributes.getValue(0).split(SPLIT_ONE);
+                if (splitStrings[0].substring(0, 7).equals(PREFIX_CONFIG)) {
+                    final String moduleName = splitStrings[0].substring(7).split(SPLIT_TWO)[0];
+                    strings[0] = moduleName;
+                    strings[1] = splitStrings[1];
+                }
+            }
+            currentTag = qName;
+        }
+
+        @Override
+        public void characters(char[] ch, int start, int length) {
+            if (currentTag != null && strings != null
+                    && strings[0] != null && currentTag.equals(nodeName)) {
+                final String currentValue = new String(ch, start, length).trim();
+                if (!currentValue.isEmpty() && !"\n".equals(currentValue)) {
+                    strings[2] = currentValue;
+                }
+                currentTag = null;
+            }
+        }
+
+        @Override
+        public void endElement(String uri, String localName, String qName) {
+            if (strings != null && strings[0] != null && qName.equals(nodeName)) {
```

Comment:
`strings != null && strings[0] != null && qName.equals(nodeName)` this check is duplicated (see previous method)

Reply:
actually, it is not duplicated and is meaningful. If I remove strings != null && strings[0] != null or  strings != null, it will both have null point error.

Reply:
done